### PR TITLE
Compose iOS accessibility support

### PR DESCRIPTION
**Pull request by zamulla**
https://github.com/JetBrains/compose-multiplatform-core/pull/1025

### CODE REVIEW

File: topics/compose/compose-ios-accessibility.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+[//]: # (title: Support for iOS accessibility features)
+
+Compose Multiplatform accessibility support for iOS allows people with disabilities to interact with Compose Multiplatform UI
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:3]:
```suggestion
Compose Multiplatform accessibility support allows people with disabilities to interact with the Compose Multiplatform UI
```

File: topics/compose/compose-ios-accessibility.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+as comfortably as with native UIs:
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:4]:
```suggestion
as comfortably as with the native iOS UI:
```

File: topics/compose/compose-ios-accessibility.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+* Screen readers and VoiceOver can access the content of the Compose Multiplatform UI.
+* Compose Multiplatform UI supports the same gestures as the native UIs for navigation and interaction.
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:6]:
```suggestion
* The Compose Multiplatform UI supports the same gestures as the native iOS UI for navigation and interaction.
```

File: topics/compose/compose-ios-accessibility.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+This is possible because semantics data produced by Compose APIs is now mapped to native objects and properties
+that are consumed by iOS Accessibility Services. For most interfaces built with Material widgets, this should happen
+automatically.
+
+You can also use this semantic data in testing and other automation: properties such as `testTag` will correctly map
+to native accessibility properties such as `accessibilityIdentifier`. This makes semantic data from Compose Multiplatform available
+to Accessibility Services and XCTest framework.
+
+Notable limitations of the current accessibility support:
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:16]: I think here you can move the repetitive part to the list's intro:

The Compose accessibility API doesn't currently cover(support):
* Live regions, announcements about..
* UIKit elements inside Compose Multiplatform UI.

File: topics/compose/compose-ios-accessibility.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+* Live regions (announcements about content changes in a specific area) are not supported yet.
+* [UIKit elements inside Compose Multiplatform UI](compose-ios-ui-integration.md#use-uikit-inside-compose-multiplatform)
+are not currently covered by the Compose accessibility API. 
+
+iOS accessibility support is in the early stages of development. If you have trouble with this feature,
+we would appreciate your feedback in the [#compose-ios](https://kotlinlang.slack.com/archives/C0346LWVBJ4/p1678888063176359)
+Slack channel, or as an issue in the [Compose Multiplatform GitHub repo](https://github.com/JetBrains/compose-multiplatform/issues).
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:23]:
```suggestion
Slack channel or as an issue in the [Compose Multiplatform GitHub repo](https://github.com/JetBrains/compose-multiplatform/issues). 
```

File: topics/compose/compose-ios-accessibility.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+are not currently covered by the Compose accessibility API. 
+
+iOS accessibility support is in the early stages of development. If you have trouble with this feature,
+we would appreciate your feedback in the [#compose-ios](https://kotlinlang.slack.com/archives/C0346LWVBJ4/p1678888063176359)
+Slack channel, or as an issue in the [Compose Multiplatform GitHub repo](https://github.com/JetBrains/compose-multiplatform/issues). 
+
+## Customize accessibility tree synchronization
+
+By default, the iOS accessibility tree is synchronized with the UI only when Accessibility Services are running;
+by default, no accessibility logs are written.
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:28]: The "accessibility logs part" doesn't seem to be relevant to the tree synchronization section. If you remove it, there wouldn't be the "by default" repetition

File: topics/compose/compose-ios-accessibility.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+You can customize this behavior, for example, when you want to debug events and interactions:
+activate the option to always synchronize the accessibility tree, so that the tree is rewritten every time the UI updates.
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:30]: The sentence is quite long, maybe split it after the colon?

File: topics/compose/compose-ios-accessibility.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+> Always synchronizing the tree can be quite useful for debugging and testing, but be aware that there is a significant
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:32]: A part of the second sentence repeats the section's intro, I think we can safely shorten this tip to:

> Constant synchronizing (or Constant synchronization of) the tree can be quite useful for debugging and testing but can degrade your app's performance.

Also, it look likes rather important, maybe turn it into a note?

File: topics/compose/compose-ios-accessibility.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    class WhenRequiredByAccessibilityServices(debugLogger: AccessibilityDebugLogger?)
+
+    // Option to always synchronize the accessibility tree
+    //
+    // You can include an AccessibilityDebugLogger to log interactions and tree syncing events
+    class Always(debugLogger: AccessibilityDebugLogger?)
+}
+```
+
+## What's next?
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:101]: That could be a list of actions as well:

* Try out the project generated by..
* Learn about  [recourse qualifiers](compose-images-resources.md) that will be helpful..

File: topics/compose/compose-ios-accessibility.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+         }
+     })
+ }) {
+    // your @Composable content
+ }
+```
+
+## Accessibility APIs
+
+The following APIs provided by Compose Multiplatform help you customize iOS accessibility behavior.
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:59]:
```suggestion
Compose Multiplatform provides the following APIs to help you customize iOS accessibility behavior.
```

File: topics/compose/compose-ios-accessibility.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+activate the option to always synchronize the accessibility tree, so that the tree is rewritten every time the UI updates.
+
+> Always synchronizing the tree can be quite useful for debugging and testing, but be aware that there is a significant
+> overhead associated with this. Synchronizing the tree with the UI regardless of Accessibility Services can degrade
+> the performance of your app.
+>
+> {type="tip"}
+
+Compose Multiplatform [provides APIs](#accessibility-apis) to configure the tree sync behavior:
+modify the `accessibilitySyncOptions` parameter inside the `configure` block of a `ComposeUIViewController` call:
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:39]:
```suggestion
modify the `accessibilitySyncOptions` parameter inside the `configure {}` block of a `ComposeUIViewController` call:
```

**Review by danil-pavlov — COMMENTED** [review]
A couple of ideas regarding the structure:

1. Consistent headers, noun-phrases "Accessibility tree synchronization/ Accessibility APIs﻿" or verbs "Customize accessibility tree synchronization/ Adjust accessibility behavior with APIs"
2. Instead of the Accessibility APIs﻿ reference section, make two use cases: Customize accessibility tree synchronization (includes AccessibilitySyncOptions description﻿) and Set up logging (includes AccessibilityDebugLogger﻿ description)

File: topics/compose/compose-ios-accessibility.md (hunk 6 above)
Comment (zamulla) [reply topics/compose/compose-ios-accessibility.md:28]: The API allows both to control the sync behavior and logging of the synchronization events, so more of a the fact that logging doesn't seem to be relevant is a bigger problem :)

I rephrased for clarity.

File: mpd.tree [hunk 12]
```diff
@@ -45,7 +45,10 @@
         <toc-element id="compose-images-resources.md"/>
         <toc-element id="compose-navigation-routing.md"/>
         <toc-element id="compose-test.md"/>
-        <toc-element id="compose-ios-ui-integration.md" toc-title="Integration with other UI frameworks"/>
+        <toc-element toc-title="iOS specific features">
```
Comment (danil-pavlov) [inline mpd.tree:48]:
```suggestion
        <toc-element toc-title="iOS-specific features">
```

File: topics/compose/compose-ios-accessibility.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+With default settings:
+* The iOS accessibility tree is synchronized with the UI only when Accessibility Services are running.
+* Synchronization events are not logged.
+
+You can customize these settings with the new Compose Multiplatform API.
+
+### Choose the tree synchronization option
+
+To debug and test events and interactions, you can change the synchronization mode:
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:34]:
```suggestion
To debug and test events and interactions, you can change the synchronization mode to:
```

File: topics/compose/compose-ios-accessibility.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+* to never sync the tree with UI (for example, to temporarily disable accessibility mapping),
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:35]:
```suggestion
* Never synchronize the tree with UI, for example, to temporarily disable accessibility mapping.
```

File: topics/compose/compose-ios-accessibility.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+* to always sync the tree, so that it is rewritten every time the UI updates (to thoroughly test the accessibility integration).
```
Comment (danil-pavlov) [inline topics/compose/compose-ios-accessibility.md:36]:
```suggestion
* Always synchronize the tree so that it is rewritten every time the UI updates to test the accessibility integration thoroughly.
```

**Review by danil-pavlov — APPROVED** [review]
